pkg/shared/emailTemplates/template_cache.go

```diff
@@ -45,17 +45,34 @@ func (t *TemplateCache) GetTextTemplate(templateName string) (*textTemplate.Temp
 }
 
 // LoadHTMLTemplateFromString parses a string into an object model and stores it in the htmlTemplateCache
-func (t *TemplateCache) LoadHTMLTemplateFromString(templateName string, raw string) error {
+func (t *TemplateCache) LoadHTMLTemplateFromString(
+	templateName string,
+	raw string,
+	embeddedTemplates map[string]string, // name to content mapping for embedded templates
+) error {
 	_, templateExists := t.htmlTemplateCache[templateName]
 	if templateExists {
 		return fmt.Errorf("htmlTemplate [%s] already exists in htmlTemplateCache", templateName)
 	}
 
-	tpl, err := htmlTemplate.New(templateName).Parse(raw)
+	// Create a new template with the specified name
+	tpl := htmlTemplate.New(templateName)
+
+	// Parse all the shared/embedded templates first
+	for name, content := range embeddedTemplates {
```

Comment:
It doesn't look like it's required to parse the template beforehand in EASi, perhaps that's because of our implementation? Have you tried without the explicit manual parsing of embedded templates? ( [easi implementation](https://github.com/CMSgov/easi-app/blob/main/pkg/email/email.go))